tox.ini

```diff
@@ -38,21 +40,24 @@ pass_env =
     CI
     TOXENV
     CODECOV_*
+set_env =
+    devdeps: UV_INDEX = https://pypi.anaconda.org/liberfa/simple https://pypi.anaconda.org/astropy/simple https://pypi.anaconda.org/scientific-python-nightly-wheels/simple
+    devdeps: UV_INDEX_STRATEGY = unsafe-any-match
 extras =
     test
+uv_resolution =
+    oldestdeps: lowest-direct
 deps =
     xdist: pytest-xdist
-    cov: pytest-cov
-    oldestdeps: minimum_dependencies
+    cov: pytest-cov >= 4.1.0
     rad: rad @ git+https://github.com/spacetelescope/rad.git@main#egg=rad
+    romancal: romancal[test] @ git+https://github.com/spacetelescope/romancal.git
+    devdeps: -r requirements-dev.txt
 commands_pre =
-    oldestdeps: minimum_dependencies roman_datamodels --filename requirements-min.txt
-    oldestdeps: pip install -r requirements-min.txt
-    devdeps: pip install -r requirements-dev.txt -U --upgrade-strategy eager
-    pip freeze
+    {list_dependencies_command}
 commands =
     pytest \
-    warnings: -W error \
+    romancal: -W ignore::DeprecationWarning -W ignore::UserWarning -W ignore::SyntaxWarning -W ignore::pytest.PytestUnknownMarkWarning -W ignore::pytest.PytestUnraisableExceptionWarning -W ignore::erfa.core.ErfaWarning -W ignore::roman_datamodels.datamodels._utils.FilenameMismatchWarning -W ignore::astropy.io.fits.verify.VerifyWarning -W ignore::RuntimeWarning --pyargs romancal \
```

Comment:
This downstream test is picking up the pytest configuration for roman_datamodels instead of the pytest configuration for romancal:
```
2024-11-22T21:24:48.5571495Z [1m============================= test session starts ==============================[0m
2024-11-22T21:24:48.5573597Z platform linux -- Python 3.12.7, pytest-8.3.3, pluggy-1.5.0
2024-11-22T21:24:48.5574444Z cachedir: .tox/romancal/.pytest_cache
2024-11-22T21:24:48.5575280Z rootdir: /home/runner/work/roman_datamodels/roman_datamodels
2024-11-22T21:24:48.5576163Z configfile: pyproject.toml
```
I'm guessing this is the need for this long list of warning filters. Would you update the configuration to use a clone of romancal instead of `pyargs`? Similar to the suggestion in https://github.com/spacetelescope/gwcs/pull/530#issuecomment-2488827411

Reply:
Its because I am using tox-uv.

Reply:
Could you elaborate? Does `tox-uv` handle warnings differently?

Reply:
This was not working for me and I didn't feel like fixing it.
https://github.com/spacetelescope/roman_datamodels/blob/ab68c3537278140607b7c290fb5fc3cf81a45cd9/tox.ini#L81

Reply:
This also uncovered the issue I fixed here: spacetelescope/romancal#1536

Reply:
Would you add some description of what the issue was? That would be helpful for whoever reviews/sees that PR.

I am concerned about maintaining this long list of warnings that are dependent on romancal. If a romancal PR adds a change that introduces a new warning (which isn't currently checked for romancal) than this job in roman_datamodels will start failing until we update this list.

If the above command (which solves the maintanence issue by using the filters of the downstream package) wasn't compatible with `tox-uv` why are we switching to it?

Reply:
Do not worry about it. I took care of it in my last push.